src/Microsoft.IdentityModel.Tokens/TokenUtilities.cs

```diff
@@ -55,5 +55,72 @@ internal static IEnumerable<SecurityKey> GetAllSigningKeys(TokenValidationParame
                 foreach (SecurityKey key in validationParameters.IssuerSigningKeys)
                     yield return key;
         }
+
+        /// <summary>
+        /// Merges claims. If a claim with same type exists in both <paramref name="claims"/> and <paramref name="subjectClaims"/>, the one in claims will be kept.
+        /// </summary>
+        /// <param name="claims"> Collection of <see cref="Claim"/>'s.</param>
+        /// <param name="subjectClaims"> Collection of <see cref="Claim"/>'s.</param>
+        /// <returns> A Merged list of <see cref="Claim"/>'s.</returns>
+        internal static IEnumerable<Claim> MergeClaims(IEnumerable<Claim> claims, IEnumerable<Claim> subjectClaims)
```

Comment:
>internal static IEnumerable<Claim> MergeClaims(IEnumerable<Claim> claims, IEnumerable<Claim> subjectClaims) [](start = 8, length = 107)

Could this be replaced by [https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.sortedset-1.unionwith?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.sortedset-1.unionwith?view=netcore-3.1)

Reply:
I took a look at UnionWith and Union.

1. UnionWith is not available on IEnumerable, to use it we need to cast it to ISet
2. Union is available but we need to provide a comparer. For that 'Claim' class need to implement IEqualityComparer<T> or IEquatable<T>